syntax = "proto3";

import "google/protobuf/timestamp.proto";

// Same contents as request from client with timestamp converted to
// protobuf format. Same contents as database PostsEntry without global_id
message NewArticle {
  string author = 1;
  string title = 2;
  string body = 3;
  google.protobuf.Timestamp creation_datetime = 4;
}

// Simple response. With global_id sent back to display on client side
message NewArticleResponse {
    enum ResultType {
        OK = 0;
        ERROR = 1;
    }

    ResultType result_type = 1;
    string global_id = 2;

    /* Should only be set if result_type is not OK. */
    string error = 3;
}

service Article {
  rpc CreateNewArticle(NewArticle) returns (NewArticleResponse);
}
